src/services/refactors/extractMethod.ts

```diff
@@ -0,0 +1,1154 @@
+                }
+                if (!substitutionsPerScope[i].has(symbolId)) {
+                    const substitution = tryReplaceWithQualifiedNameOrPropertyAccess(symbol.exportSymbol || symbol, scope, isTypeName);
+                    if (substitution) {
+                        substitutionsPerScope[i].set(symbolId, substitution);
+                    }
+                    else if (isTypeName) {
+                        errorsPerScope[i].push(createDiagnosticForNode(identifier, Messages.TypeWillNotBeVisibleInTheNewScope));
+                    }
+                    else {
+                        usagesPerScope[i].usages.set(identifier.text as string, { usage, symbol, node: identifier });
+                    }
+                }
+            }
+            return symbolId;
+        }
+
+        function checkForUsedDeclarations(node: Node) {
+            // If this node is entirely within the original extraction range, we don't need to do anything.
+            if (node === targetRange.range || (isArray(targetRange.range) && targetRange.range.indexOf(node as Statement) >= 0)) {
+                return;
+            }
+
+            // Otherwise check and recurse.
+            const sym = checker.getSymbolAtLocation(node);
+            if (sym && visibleDeclarationsInExtractedRange.some(d => d === sym)) {
+                for (const scope of errorsPerScope) {
+                    scope.push(createDiagnosticForNode(node, Messages.CannotExtractExportedEntity));
+                }
+                return true;
+            }
+            else {
+                forEachChild(node, checkForUsedDeclarations);
+            }
+        }
+
+        function tryReplaceWithQualifiedNameOrPropertyAccess(symbol: Symbol, scopeDecl: Node, isTypeNode: boolean): PropertyAccessExpression | EntityName {
+            if (!symbol) {
+                return undefined;
+            }
+            if (symbol.getDeclarations().some(d => d.parent === scopeDecl)) {
+                return createIdentifier(symbol.name);
+            }
+            const prefix = tryReplaceWithQualifiedNameOrPropertyAccess(symbol.parent, scopeDecl, isTypeNode);
+            if (prefix === undefined) {
+                return undefined;
+            }
+            return isTypeNode ? createQualifiedName(<EntityName>prefix, createIdentifier(symbol.name)) : createPropertyAccess(<Expression>prefix, symbol.name);
+        }
+    }
+
+    function getParentNodeInSpan(node: Node, file: SourceFile, span: TextSpan): Node {
+        while (node) {
+            if (!node.parent) {
+                return undefined;
+            }
+            if (isSourceFile(node.parent) || !spanContainsNode(span, node.parent, file)) {
+                return node;
+            }
+
+            node = node.parent;
+        }
+    }
+
+    function spanContainsNode(span: TextSpan, node: Node, file: SourceFile): boolean {
+        return textSpanContainsPosition(span, node.getStart(file)) &&

```

Comment:
>textSpanContainsPosition(span, node.getStart(file))  [](start = 15, length = 52)

Why not just compare the start positions?  (Both for symmetry and to avoid the unnecessary comparison between the node start and the span end.)